### ISSUE TITLE
unexpected end of JSON input

### ISSUE THREAD

**Issue by david60** [NebulousLabs/Sia#1288]
Hi,

I noticed a few errors like:

2016/06/01 13:38:07.311245 update.go:296: unexpected end of JSON input

which is from within ProcessConsensusChange. I believe this is because of lookups of storageobligations which have been deleted, but for which there are still action items. e.g. possibly a check like:

```go
@@ -269,6 +278,9 @@ func (h *Host) ProcessConsensusChange(cc modules.ConsensusChange) {
                        bso := tx.Bucket(bucketStorageObligations)
                        for _, soid := range obligationIDs {
                                soBytes := bso.Get(soid[:])
+                               if soBytes == nil {
+                                       continue
+                               }
```

may help. This is might be a useful condition to catch, because I believe (but didn't check) an error here would abort the bolt db transaction and lose previous changes.

David

**Comment by DavidVorick** [comment]
Was this happening without the double-subscribe error? I can see why being doubly subscribed to the consensus set would cause this.

**Comment by DavidVorick** [comment]
Ok, it looks like this could trigger in a singly subscribed host as well. Most of the errors are handled by skipping to the next iteration of whatever loop. I think `return err` was a mistake. Addressed in #1291

**Comment by DavidVorick** [comment]
If the host were constructed perfectly, there should never be multiple action items for the same storage obligation at the same height. Currently I'm pretty sure that's not true, which is why we sometimes see this error. I don't think it would be very easy though to fix up the action item scheduling.
